### ISSUE TITLE
Create implementation plan for Cred.ly

### ISSUE THREAD

**Issue by hannahkane** [mozilla/teach.webmaker.org#710]
We are considering offering Learning Networks badges through Cred.ly. 

Work is being done now to finalize the decision, but in the meantime we need to create an implementation plan that accounts for the following product requirements:
 * Seamless account creation/authentication experience for Teach users (that is, Webmaker users)
 * Badge creation and issuing experience, initially only for staff members
 * Landing page that displays all of our badge offerings
 * Seamless badge application and badge receipt workflows, including branded notifications
 * Claim code creation and redemption experience
 * Seamless "share" workflow for displaying badges on social media
 * Displaying of earned badges on user profiles (Note: these haven't been built yet, so this will be a plan for future implementation)
 * Reporting functionality

**Comment by hannahkane** [comment]
cc @simonwex who might have detail regarding what should go in the implementation plan.

**Comment by hannahkane** [comment]
@simonwex - can you add any necessary detail about the expected outcome of this ticket? I'm putting it in the next heartbeat, so would be great to get clarity before Monday.

**Comment by hannahkane** [comment]
Slightly more user story-fied beginnings of statement of work here: https://etherpad.mozilla.org/badgetix

**Comment by toolness** [comment]
I am investigating this now by familiarizing myself with the cred.ly API.

@simonwex do we already have a mofo account with credly? I've created a personal account w/ a sandbox app for now, but if we have a mofo account it'd probably be better for me to use that, since it looks like there are quite a few [limitations](https://credly.com/upgrade) to the free accounts.

**Comment by toolness** [comment]
Hmm, I posted this on the etherpad, but thought I'd mention it here too.

The [Credly Best Practices for Account Verification](https://developers.credly.com/best-practices#verification) states:

> Open Credit requires all members to verify their email addresses before  they can use them to give or receive credit. Due to this restriction,  app developers implementing a user registration workflow should note  that before a new account can give a badge or accept a badge, the new  user must click the verification link sent to the email address they  registered with and complete the registration flow on credly.com.

This seems to imply that users who want to use Webmaker and receive badges may need to verify their email *twice*: once so *we* know their email is verified, and once so cred.ly knows it's verified. I suspect we have very little control over the registration flow on credly.com. (We could alternatively just trust credly's belief that the user's email is verified, but we'd have to talk to the id.webmaker.org team about that.)

Also, it seems like Credly might have a one-to-one mapping between user accounts and email addresses, which Webmaker login does not have, as explained in https://github.com/mozilla/id.webmaker.org/issues/120#issuecomment-88969419. So that might also cause complications down the road.

**Comment by toolness** [comment]
Hey, I just talked to @simonwex and it looks like we might actually be able to *change* the Credly API, which would be cool. So we might be able to alter this aspect of things.

**Comment by hannahkane** [comment]
I'd originally had "seamless authentication experience" as a requirement for this project, though I think we might reconsider if it truly weren't an option. It may not tip the scales in any significant way.

Let's verify the option to change the API.

**Comment by toolness** [comment]
A few more notes:

* I verified that there is indeed a one-to-one mapping between user accounts and email addresses in Credly, yet a [many-to-one](https://github.com/mozilla/id.webmaker.org/issues/120#issuecomment-88969419) mapping in Webmaker, which could significantly complicate things.
* The badge application admin UI is pretty bare-bones. After rejecting a badge application and specifying a reason, I was unable to find the rejected application (or rejection reason) anywhere in the UI, and even poking around in the API for a bit didn't yield anything either. (In fact, I didn't even receive any notification on the account I applied from, so it seems like the rejection went into a black hole.) There aren't currently any user stories in our [badgetix etherpad](https://etherpad.mozilla.org/badgetix) that require this, but I suspect it'd eventually be useful to have--back when working on Chicago Summer of Learning, we envisioned a badge application workflow that worked much like a GitHub Issue, e.g. where it was really more of a conversation with a recorded history than a one-time rejection/acceptance. That said, perhaps Credly could improve this for us and their users.
